src/scales/scale.time.js

```diff
@@ -175,8 +175,20 @@ module.exports = function(Chart) {
 		var ticks = [];
 		if (options.maxTicks) {
 			var stepSize = options.stepSize;
-			ticks.push(options.min !== undefined ? options.min : niceRange.min);
-			var cur = moment(niceRange.min);
+			var startTick = options.min !== undefined ? options.min : niceRange.min;
+			var majorUnitStart = startTick;
+			if (options.majorUnit) {
+				majorUnitStart = moment(startTick).add(1, options.majorUnit).startOf(options.majorUnit);
+			}
+			var startRange = majorUnitStart.valueOf() - startTick;
+			var startFraction = startRange % (interval[options.unit].size * stepSize);
+			var alignedTick = startTick;
+			ticks.push(startTick);
+			if (startTick !== alignedTick + startFraction && options.majorUnit && !options.timeOpts.round && !options.timeOpts.isoWeekday) {
```

Comment:
If `startTick` is not equal to time that aligned by `stepSize` accordingly to `majorUnit` we need to find first aligned tick so then we can push ticks with `stepSize` and there will be ticks aligned to `majorUnit`.
In other words if we have:
`startTick` == 20:12;
`majorUnit` == "hour";
`sterSize` == "3 hour";
`majorUnitStart` will equal to "00:00", first aligned tick will equal to "21:00" and `startFraction` will equal to "48 minutes".
With this value of `startFraction` we can determine `alignedTick` (tick which aligned by `stepSize` accordingly to `majorUnit` (21:00, 00:00, 03:00 etc..))
When we perform `startTick !== alignedTick + startFraction` at line #187 we check if we need to push aligned tick (21:00) or `startTick` (20:12) is already aligned.
So `startFraction` is not difference between `startTick` (20:12) and `majorUnitStart` (00:00) but difference between `startTick` (20:12) and first aligned tick (21:00)